lib/r10k/module/stub.rb

```diff
@@ -0,0 +1,103 @@
+require 'r10k/module'
+
+class R10K::Module::Stub < R10K::Module::Base
```

Comment:
I agree that `Stub` isn't quite right. Makes this almost sound like a testing implementation, rather than something we actually use.

I see what you mean about the issues with putting it into the base module class. Are you thinking that if you had module implementations "register" a version parsing schema, we could do away with a separate class altogether (and therefore don't need to think of a name)?

If we want to keep this class... we could call it something like `metadata`? To imply that it's just the external properties of the module and not an actual module?